packages/flutter/test/material/page_test.dart

```diff
@@ -22,46 +40,45 @@ void main() {
 
-    FadeTransition widget2Opacity =
-        tester.element(find.text('Page 2')).findAncestorWidgetOfExactType<FadeTransition>()!;
-    Offset widget2TopLeft = tester.getTopLeft(find.text('Page 2'));
-    final Size widget2Size = tester.getSize(find.text('Page 2'));
+    ScaleTransition widget1Scale = _findForwardScaleTransition(find.text('Page 1'));
+    ScaleTransition widget2Scale = _findForwardScaleTransition(find.text('Page 2'));
+    FadeTransition widget2Opacity = _findForwardFadeTransition(find.text('Page 2'));
 
-    // Android transition is vertical only.
-    expect(widget1TopLeft.dx == widget2TopLeft.dx, true);
-    // Page 1 is above page 2 mid-transition.
-    expect(widget1TopLeft.dy < widget2TopLeft.dy, true);
-    // Animation begins 3/4 of the way up the page.
-    expect(widget2TopLeft.dy < widget2Size.height / 4.0, true);
-    // Animation starts with page 2 being near transparent.
-    expect(widget2Opacity.opacity.value < 0.01, true);
+    // Page 1 is enlarging, starts from 1.0.
+    expect(widget1Scale.scale.value, greaterThan(1.0));
+    // Page 2 is enlarging from the value less than 1.0.
+    expect(widget2Scale.scale.value, lessThan(1.0));
+    // Page 2 is becoming none transparent.
+    expect(widget2Opacity.opacity.value, lessThan(1.0));
 
-    await tester.pump(const Duration(milliseconds: 300));
+    await tester.pump(const Duration(milliseconds: 250));
+    await tester.pump(const Duration(milliseconds: 1));
 
     // Page 2 covers page 1.
     expect(find.text('Page 1'), findsNothing);
     expect(find.text('Page 2'), isOnstage);
 
     tester.state<NavigatorState>(find.byType(Navigator)).pop();
     await tester.pump();
-    await tester.pump(const Duration(milliseconds: 1));
+    await tester.pump(const Duration(milliseconds: 100));
 
-    widget2Opacity =
-        tester.element(find.text('Page 2')).findAncestorWidgetOfExactType<FadeTransition>()!;
-    widget2TopLeft = tester.getTopLeft(find.text('Page 2'));
+    widget1Scale = _findForwardScaleTransition(find.text('Page 1'));
+    widget2Scale = _findForwardScaleTransition(find.text('Page 2'));
+    widget2Opacity = _findForwardFadeTransition(find.text('Page 2'));
 
-    // Page 2 starts to move down.
-    expect(widget1TopLeft.dy < widget2TopLeft.dy, true);
-    // Page 2 starts to lose opacity.
-    expect(widget2Opacity.opacity.value < 1.0, true);
+    // Page 1 is narrowing down, but still larger than 1.0.
+    expect(widget1Scale.scale.value, greaterThan(1.0));
+    // Page 2 is smaller than 1.0.
+    expect(widget2Scale.scale.value, lessThan(1.0));
+    // Page 2 is becoming transparent.
+    expect(widget2Opacity.opacity.value, lessThan(1.0));
 
-    await tester.pump(const Duration(milliseconds: 300));
+    await tester.pump(const Duration(milliseconds: 200));
+    await tester.pump(const Duration(milliseconds: 1));
 
     expect(find.text('Page 1'), isOnstage);
     expect(find.text('Page 2'), findsNothing);
```

Comment:
might be better to have two versions of this test, one with the old transitions and one with the new transitions. the old test shouldn't need to change at all except for being explicit about the transitions, right?

Reply:
Exactly.

Reply:
Sure

Reply:
Oh, it's a bit confusing here. I do keep the original version of the test with a new name here.